CHANGELOG.md

```diff
@@ -85,6 +85,15 @@ Adding a new version? You'll need three changes:
   [#4749](https://github.com/Kong/kubernetes-ingress-controller/pull/4749)
 - Removed Knative support.
   [#4748](https://github.com/Kong/kubernetes-ingress-controller/pull/4748)
+- Removed support for the `debug-log-reduce-redundancy` CLI flag.
+  [#4688](https://github.com/Kong/kubernetes-ingress-controller/pull/4688)
+- The "text" logging format has changed. "json" should be used for
+  machine-parseable logs.
+  [#4688](https://github.com/Kong/kubernetes-ingress-controller/pull/4688)
+- The "warn", "fatal", and "panic" log levels are no longer available. "error"
+  is now the highest log level. "warn" logs are now logged at "info" level, but
+  include an additional "warning=true" field.
```

Comment:
> "warn" logs are now logged at "info" level, but include an additional "warning=true" field

What's the mechanism for that? 🤔 Could we "support" the warning log level? For example by introducing our own logger interface that embeds the `logr.Logger` and adds a warning level (and appropriate method)? Obviously we'd do that separately but I'm just wondering since adding a key value pair to the log entry is somewhat orthogonal to the logging level.

Reply:
https://github.com/Kong/kubernetes-ingress-controller/commit/572c2f448d53db90e3c4a9c988a6a965168d0b88 is the mechanism.

Levels are now verbosity values by convention, and controller-runtime has no less verbose level than what's effectively info. It has some things that are warning-ish at info, and some others that mention it in the message body: https://gist.github.com/rainest/e55fe44e54e4e1ab55d56f1210afb144#file-controller_runtime_logs-txt

I don't think there's any way to give controller-runtime a logger that bumps verbosity by one if we wanted to use a 0=warning, 1=info, 2=debug convention for our own logs.

On review of what we [were logging at warning](https://gist.github.com/rainest/e55fe44e54e4e1ab55d56f1210afb144#file-existing_kic_warnings-txt), it looks like they're all reasonable to just bump to error (none of them are things that should happen), so e1fd0608af7228fb62927cb0ee9815b1f8766fec

Reply:
Given that all warning logs where converted to error now I'm 👍 this change.

If it ever comes to a point where we need warning logs, we can solve that then.